src/components/annotations/AnnotationVariantItem.vue

```diff
@@ -1,23 +1,140 @@
 <template>
-    <div v-for="(variant, i) in annotation.body.value" :key="i"  class="t-items-center t-flex">
-          <span v-if="variant.witness" v-html="variant.witness" class="t-mr-7"/>
-          <span v-else  class="t-mr-7"> - </span>
-          <span v-html="variant.entry"/>
+  <div
+    v-for="(variant, i) in annotation.body.value"
+    :key="i"
+    class="t-items-center t-flex t-mb-1 t-relative"
+    :class="[
+      't-py-2 t-px-3 t-mb-1 t-rounded-md',
+      { 'hover:t-bg-gray-200 dark:hover:t-bg-gray-600 t-cursor-pointer': !isText(annotation) && !isActive(annotation) },
+      { 't-bg-gray-300 dark:t-bg-gray-600 active': isVariantItemActive(variant.witness)}]"
+    :data-annotation-id="annotation.id"
+    @click="isText(annotation) ? ()=>{} : handleClick(variant.witness, i)"
+  >
+    <div
+      class="t-relative  t-rounded-3xl t-box-border t-w-75 t-h-8 t-border-2 t-p-[2px]"
+      :style="{'border-color': getItemColorBasedOnIndex(i)}"
+    >
+      <span
+        v-if="variant.witness"
+        class="t-text-sm"
+        v-html="variant.witness"
+      />
+      <span
+        v-else
+        class="t-text-sm"
+      > - </span>
     </div>
+    <span
+      class="t-absolute t-right-[50%]"
+      v-html="variant.entry"
+    />
+  </div>
 </template>
 
 
 
 <script setup lang="ts">
+import { getItemColorBasedOnIndex } from '@/utils/color';
+import { reactive } from 'vue';
+import * as AnnotationUtils from '@/utils/annotations';
+
 
 export interface Props {
-  annotation: Annotation
+  annotation: Annotation,
+  isText: (annotation: Annotation) => boolean,
+  isActive: (annotation: Annotation) => boolean,
+  toggle: (annotation: Annotation) => void,
 }
 
 const props = withDefaults(defineProps<Props>(), {
   annotation: () => <Annotation>{},
+  isText: () => true,
+  isActive: () => true,
+  toggle: () => null,
 })
 
+let variantItemsSelection = initializeSelectionOfVariantItem()
```

Comment:
I am not sure why we need to `watch` to initialize the `variantItemsSelection`, since the props.annotation will not change during the code execution in this, will it ? For each annotation we get, the variant items list is created, so I would assume no more change in the props.annotation in this case... ?

Reply:
In this case not, but it's just not a good practice to define reactive variables in functions and makes the component vulnarable.

Reply:
alright, I understand, thanks